Add support of AOSC OS and loongson architecture

4Darmygeometry opened the pull request:
AOSC OS uses oma to install and manage the package. It can be see in https://aosc.io/oma .
Loongson Architecture, which also meant LoongArch, is an architecture which is powered by Loongson Technology LLC.

4Darmygeometry committed:
Add support of AOSC OS and loongson architecture

AOSC OS uses oma to install and manage the package. It can be see in https://aosc.io/oma .
Loongson Architecture, which also meant LoongArch, is an architecture which is powered by Loongson Technology LLC.

4Darmygeometry commented:
AOSC OS（安同OS）使用oma软件包管理器管理软件包，且部分软件包名与debian系拼写存在区别，不适合直接使用debian系的apt安装前置依赖。现改变脚本内容让安装脚本适配AOSC OS（安同OS）通过oma安装软件包的方式。可见介绍：https://aosc.io/oma
龙架构，是龙芯中科自主研发的处理器架构，Steam++编译所需.net 9与vscodium已有上游化支持。龙架构的龙芯可以使用LATX转译+Wine运行Windows版的Steam。本次脚本添加对LoongArch64的判定。

4Darmygeometry committed:
Update README.md

添加龙芯版构建声明

4Darmygeometry committed:
Add LoongArch64 build script

Add LoongArch64 build script

4Darmygeometry committed:
Add LoongArch64 Library Support

AigioL commented:
除了网络加速插件外其他的插件很多依赖 P/Invoke 操作 Steam 客户端，代码里使用不同的 tfm 区分了 Windows 与 Linux 的行为，对于在 Wine 中运行的 Steam，应该都不起作用，而 Windows 版 Steam++ 依赖高版本 Win10SDK，以及需要一些 WinRT API 与系统 exe 等内容，应该在 Wine 中会无法运行或出现错误

编译 linux-loongarch64 版可以在龙芯上本机原生运行网络加速用的代理，性能要比转译的更好
可以加一个 rid 顺便编译，不过目前用的自己写的控制台程序进行发布操作的，发布工具的项目需要修改一下以支持新的 rid
https://github.com/BeyondDimension/SteamTools/tree/develop/src/BD.WTTS.Client.Tools.Publish

4Darmygeometry commented:
> 除了网络加速插件外其他的插件很多依赖 P/Invoke 操作 Steam 客户端，代码里使用不同的 tfm 区分了 Windows 与 Linux 的行为，对于在 Wine 中运行的 Steam，应该都不起作用，而 Windows 版 Steam++ 依赖高版本 Win10SDK，以及需要一些 WinRT API 与系统 exe 等内容，应该在 Wine 中会无法运行或出现错误
> 
> 编译 linux-loongarch64 版可以在龙芯上本机原生运行网络加速用的代理，性能要比转译的更好
> 可以加一个 rid 顺便编译，不过目前用的自己写的控制台程序进行发布操作的，发布工具的项目需要修改一下以支持新的 rid
> https://github.com/BeyondDimension/SteamTools/tree/develop/src/BD.WTTS.Client.Tools.Publish

那先把我AOSC OS的oma支持先并入，谢谢。
Steam++可以加速github，这个是龙芯所需要的。

4Darmygeometry committed:
标记统一使用.NET 9.0使得龙芯新世界代码与主线一致

由于.NET 9.0才在龙芯中构建主线新世界版本，为保证龙芯版与主线代码尽可能保持一致，本项目建议使用.NET 9.0与C# 13。
完善操作系统相关要求并补充最低CPU要求。

4Darmygeometry committed:
Update README.md

4Darmygeometry committed:
Update README.en.md

与中文的保持同步

4Darmygeometry committed:
Update README.md

添加对于兆芯和飞腾使用该程序描述。

4Darmygeometry committed:
Update README.md

完善编译定义

4Darmygeometry committed:
Update open-source-library.md

4Darmygeometry committed:
Update open-source-library.md

4Darmygeometry commented:
> 除了网络加速插件外其他的插件很多依赖 P/Invoke 操作 Steam 客户端，代码里使用不同的 tfm 区分了 Windows 与 Linux 的行为，对于在 Wine 中运行的 Steam，应该都不起作用，而 Windows 版 Steam++ 依赖高版本 Win10SDK，以及需要一些 WinRT API 与系统 exe 等内容，应该在 Wine 中会无法运行或出现错误
> 
> 编译 linux-loongarch64 版可以在龙芯上本机原生运行网络加速用的代理，性能要比转译的更好
> 可以加一个 rid 顺便编译，不过目前用的自己写的控制台程序进行发布操作的，发布工具的项目需要修改一下以支持新的 rid
> https://github.com/BeyondDimension/SteamTools/tree/develop/src/BD.WTTS.Client.Tools.Publish

![1736176776508.png](https://github.com/user-attachments/assets/24ee9c9e-908f-4ce5-9fda-3eceebf216c5)

![1736176787586.png](https://github.com/user-attachments/assets/61aa824d-1523-4d52-9dea-772464299146)

Steam++龙架构移植基本完成，可启动主界面；整理后文件夹版本可以启动插件@rmbadmin

4Darmygeometry committed:
Merge branch 'BeyondDimension:develop' into develop

4Darmygeometry committed:
Update README.md

4Darmygeometry committed:
Update README.en.md

4Darmygeometry committed:
添加libpng前置依赖所需的loongarch64 c文件

Mossimos committed:
✨ 发布工具支持 RiscV64、, LoongArch64 发布

4Darmygeometry committed:
添加龙芯3A6000电脑原生运行截图

添加龙芯3A6000电脑适配该软件后原生运行截图

4Darmygeometry committed:
Update README.md

添加龙芯3A6000电脑原生运行截图

4Darmygeometry commented:
龙架构主程序基本适配完成
![linux-3A6000-homepage](https://github.com/user-attachments/assets/827eeee0-5647-4097-9a17-bdfd8538de1d)
![linux-3A6000-script](https://github.com/user-attachments/assets/c33ba2f0-3cd0-4be2-a36e-1a0948e99a71)
![linux-3A6000-ASF](https://github.com/user-attachments/assets/844b9262-6c96-4504-9967-fcde4599d557)
![linux-3A6000-settings](https://github.com/user-attachments/assets/0d8b7444-69cf-4986-af14-77daa556e101)

4Darmygeometry committed:
添加龙芯3A6000电脑加速github截图

4Darmygeometry committed:
添加龙芯nuget源注意事项

4Darmygeometry committed:
删除次要截图

4Darmygeometry committed:
删除次要截图

4Darmygeometry committed:
Add English screenshots

4Darmygeometry committed:
Update README.en.md

4Darmygeometry committed:
Add English Support

4Darmygeometry committed:
Update online_install.sh

4Darmygeometry committed:
支持AOSC OS

4Darmygeometry committed:
支持中文和其他语言的的桌面文件夹

4Darmygeometry commented:
现龙架构软件包已完全可用并可复现编译

4Darmygeometry committed:
更新 README.md

4Darmygeometry committed:
Merge branch 'BeyondDimension:develop' into develop

4Darmygeometry committed:
更新判断发行版方式测试文件

4Darmygeometry committed:
通过/etc/os-release形式判断发行版并支持slackware等发行版包格式

通过/etc/os-release形式判断发行版，并支持gentoo、slackware、opensuse、mageia、solus linux等发行版包格式，支持判断loongnix25特殊情况

4Darmygeometry committed:
Update README.md

4Darmygeometry committed:
紧急修改sh错误

4Darmygeometry committed:
Update README.md

4Darmygeometry committed:
Add files via upload

4Darmygeometry committed:
Add files via upload

4Darmygeometry committed:
Delete loongson-build.md

4Darmygeometry committed:
Update README.md

4Darmygeometry committed:
把AOSC OS放在前面

4Darmygeometry committed:
纠正中科方德安装包错误并支持勇豹系统

纠正中科方德安装包错误并支持勇豹系统，安装包进度条实时显示解压进度，bug fixed

4Darmygeometry committed:
脚本已无需依赖pv，去掉pv安装选项

4Darmygeometry commented:
现脚本已可用并支持yongbao、slackware、gentoo等系统

4Darmygeometry committed:
Update README.md

driver1998 commented on README.en.md:
```diff
@@ -79,6 +100,31 @@ Read what we [milestones](https://github.com/BeyondDimension/SteamTools/mileston
+	- Desktop applications
+		- .NET desktop development
+
+### Due to the fact that the new world NuGet has not yet been fully upstreamed, the LoongArch64 version of this project uses the NuGet source provided by [Loongson Open Source Community](https://nuget.loongnix.cn) to provide SkiaSharp, HarfBuzzSharp, etc. This also allows compatibility with the three old world distributions: Loongnix 20, UOS 20, and Kylin.
```
I don't think https://nuget.loongnix.cn provides new world builds of libskiasharp and harfbuzzsharp, are you sure the resulting build runs fine on new world systems?

4Darmygeometry replied:
@Catty2014

4Darmygeometry replied:
> I don't think https://nuget.loongnix.cn provides new world builds of libskiasharp and harfbuzzsharp, are you sure the resulting build runs fine on new world systems?

https://github.com/mono/SkiaSharp/pull/3198
Since Mar 25, 2025, The LoongArch New World Support of skiasharp and harfbuzzsharp had been merged into the main line.
自从2025年3月25日，SkiaSharp和HarfBuzzSharp的龙架构新世界支持已经主线化，不用再从龙芯nuget源拉取文件了。

4Darmygeometry committed:
恢复回原来的readme

4Darmygeometry committed:
Delete doc/loongson-build.md

4Darmygeometry committed:
Delete res/screenshots/linux-3A6000-ASF-en.png

4Darmygeometry committed:
Delete res/screenshots/linux-3A6000-ASF.png

4Darmygeometry committed:
Delete res/screenshots/linux-3A6000-accelerate-en.png

4Darmygeometry committed:
Delete res/screenshots/linux-3A6000-accelerate.png

4Darmygeometry committed:
Delete res/screenshots/linux-3A6000-settings-en.png

4Darmygeometry committed:
Delete res/screenshots/linux-3A6000-settings.png

4Darmygeometry committed:
Merge branch 'BeyondDimension:develop' into develop

4Darmygeometry committed:
尝试解决分支冲突

4Darmygeometry committed:
Merge branch 'develop' into develop

4Darmygeometry committed:
重新添加龙架构判断

4Darmygeometry committed:
更新龙架构编译脚本

4Darmygeometry committed:
更新龙架构构建脚本

4Darmygeometry committed:
dotnet 9.0

4Darmygeometry committed:
update mac script to dotnet 9.0

4Darmygeometry committed:
Merge branch 'BeyondDimension:develop' into develop

4Darmygeometry committed:
Merge branch 'BeyondDimension:develop' into develop

4Darmygeometry committed:
Merge branch 'BeyondDimension:develop' into develop

4Darmygeometry committed:
Merge branch 'BeyondDimension:develop' into develop

4Darmygeometry committed:
解决LoongArchLinux及缺少wget发行版的下载问题

解决LoongArchLinux因为版本标识为rolling导致下载不了Watt Toolkit的问题
以及龙架构deepin及LoongArchLinux缺少wget下载Watt Toolkit报未知的最新版本问题

4Darmygeometry committed:
Merge branch 'BeyondDimension:develop' into develop

4Darmygeometry committed:
Update Program.cs

由于补丁https://github.com/mono/SkiaSharp/commit/e5f74e7db54c7225297bbec62c7de89c4212be2b ，SkiaSharp已经可以在RISC-V 64直接编译

4Darmygeometry committed:
Update Program.cs

Add a comma

4Darmygeometry committed:
Merge branch 'BeyondDimension:develop' into develop

4Darmygeometry committed:
Merge branch 'BeyondDimension:develop' into develop

4Darmygeometry committed:
Merge branch 'BeyondDimension:develop' into develop

4Darmygeometry committed:
Merge branch 'BeyondDimension:develop' into develop

4Darmygeometry committed:
Merge branch 'BeyondDimension:develop' into develop

4Darmygeometry committed:
Program.cs:SkiaSharp的龙架构和RISC-V支持已经主线化了

https://github.com/mono/SkiaSharp/pull/3192
https://github.com/mono/SkiaSharp/pull/3194
https://github.com/mono/SkiaSharp/pull/3198
https://github.com/mono/SkiaSharp/pull/3200

4Darmygeometry committed:
使用/p:PublishTrimmed=true压缩龙架构二进制文件体积

https://www.cnblogs.com/51net/p/16046743.html

4Darmygeometry committed:
在Linux脚本中添加32位系统直接退出选项

Watt Toolkit已经不支持32位，故在Linux脚本中添加32位系统直接退出选项

4Darmygeometry committed:
之前放错脚本了

4Darmygeometry committed:
Update and rename testcommand.sh to Linux.sh

Linux.sh为online_install.sh的压缩版，此处作测试命令

4Darmygeometry committed:
添加扩展指令判断脚本

4Darmygeometry committed:
更新指令集判断脚本

4Darmygeometry committed:
添加对系统Glibc库还是musl库的判断

目前Watt Toolkit Linux版仅支持使用Glibc库的Linux发行版

4Darmygeometry committed:
Update Linux.sh

4Darmygeometry committed:
CRLF to LF

4Darmygeometry committed:
Add copy command

4Darmygeometry committed:
修正安装脚本报错

4Darmygeometry committed:
WattToolkit.sln:Update it to support RV64 and LA64

WattToolkit.sln:Update it to support RISC-V64 and LoongArch64, and eliminate 32-bit support

4Darmygeometry committed:
CRLF TO LF

4Darmygeometry committed:
RV64指令集支持改进

4Darmygeometry committed:
Create rv64_build.sh

4Darmygeometry committed:
添加对中标麒麟（使用dnf）与开放麒麟（使用apt）的支持

4Darmygeometry committed:
修正脚本错误

4Darmygeometry committed:
Merge branch 'BeyondDimension:develop' into develop

4Darmygeometry committed:
Merge branch 'develop' into develop

4Darmygeometry committed:
更新sln与主线保持一致

4Darmygeometry committed:
Update .NET publish commands for new framework version

4Darmygeometry committed:
update to net10.0

4Darmygeometry committed:
lf backspace

4Darmygeometry committed:
Update .NET publish targets to net10.0

4Darmygeometry committed:
Update target framework to net10.0 in publish script

4Darmygeometry committed:
Merge branch 'BeyondDimension:develop' into develop

4Darmygeometry commented:
> 除了网络加速插件外其他的插件很多依赖 P/Invoke 操作 Steam 客户端，代码里使用不同的 tfm 区分了 Windows 与 Linux 的行为，对于在 Wine 中运行的 Steam，应该都不起作用，而 Windows 版 Steam++ 依赖高版本 Win10SDK，以及需要一些 WinRT API 与系统 exe 等内容，应该在 Wine 中会无法运行或出现错误
> 
> 编译 linux-loongarch64 版可以在龙芯上本机原生运行网络加速用的代理，性能要比转译的更好 可以加一个 rid 顺便编译，不过目前用的自己写的控制台程序进行发布操作的，发布工具的项目需要修改一下以支持新的 rid https://github.com/BeyondDimension/SteamTools/tree/develop/src/BD.WTTS.Client.Tools.Publish

现在龙芯已经可以运行Linux版Steam了，提供了前置加速条件
https://www.bilibili.com/video/BV1Cay9BKEtf/

4Darmygeometry committed:
Merge branch 'BeyondDimension:develop' into develop

4Darmygeometry committed:
Change target framework from net472 to net10.0

解决architecture不支持LoongArch64和RiscV64的问题

4Darmygeometry committed:
Merge branch 'BeyondDimension:develop' into develop

4Darmygeometry committed:
Merge branch 'BeyondDimension:develop' into develop

4Darmygeometry committed:
Merge branch 'BeyondDimension:develop' into develop

4Darmygeometry committed:
Merge branch 'BeyondDimension:develop' into develop

4Darmygeometry committed:
Update SkiaSharp and HarfBuzzSharp versions in build script

4Darmygeometry committed:
Update SkiaSharp and HarfBuzzSharp native assets

4Darmygeometry committed:
Merge branch 'BeyondDimension:develop' into develop

4Darmygeometry committed:
Merge branch 'BeyondDimension:develop' into develop

4Darmygeometry committed:
解决程序在.NET存在时找不到.NET的问题

4Darmygeometry committed:
解决Windows下兼容性问题

4Darmygeometry committed:
优化脚本构建逻辑

Updated library versions and added scripts for Linux build.

4Darmygeometry commented:
![Image_1772346760431.png](https://github.com/user-attachments/assets/1e157d23-4a08-4365-91d4-9b6a6daf5a57)

![Image_1772346764338.png](https://github.com/user-attachments/assets/180d9354-cdb9-4505-b52e-bfe72567ee70)

龙芯3A6000上已经实现了Watt Toolkit对Linux版Steam的本地反代加速与库存管理之类
https://www.bilibili.com/video/BV1oYf4BDEer/

4Darmygeometry committed:
Merge branch 'BeyondDimension:develop' into develop

4Darmygeometry committed:
换源，后续更新至最新版以获得更好的Linux挂卡体验

Updated the URL and branch for the ArchiSteamFarm submodule.

4Darmygeometry committed:
Merge branch 'BeyondDimension:develop' into develop

4Darmygeometry committed:
chore: update ArchiSteamFarm submodule to latest main branch

4Darmygeometry committed:
Update native asset versions for Loongarch build

4Darmygeometry committed:
Update SkiaSharp and HarfBuzzSharp native assets versions

4Darmygeometry committed:
Enhance build script with package versioning and structure

Updated the build script to include explicit package versioning and improved output directory structure.

4Darmygeometry committed:
Configure Loongarch NuGet source and update packages

Added configuration for Loongarch NuGet source and updated package versions for SkiaSharp, HarfBuzzSharp, and SQLite. Enhanced the publishing process for the main application and plugins, including native library handling.

4Darmygeometry committed:
Stable化

4Darmygeometry committed:
Stable化
